e3sm_diags/driver/tc_analysis_driver.py

```diff
@@ -163,13 +160,40 @@ def generate_tc_metrics_from_te_stitch_file(te_stitch_file: str) -> Dict[str, An
     """
     logger.info("\nGenerating TC Metrics from TE Stitch Files")
     logger.info("============================================")
+    if not os.path.exists(te_stitch_file):
+        raise FileNotFoundError(f"The file {te_stitch_file} does not exist.")
+
     with open(te_stitch_file) as f:
-        lines = f.readlines()
+        lines_orig = f.readlines()
+
+    if not lines_orig:
+        raise ValueError(f"The file {te_stitch_file} is empty.")
+
+    line_ind = []
+    data_start_year = int(te_stitch_file.split(".")[-2].split("_")[-2])
+    data_end_year = int(te_stitch_file.split(".")[-2].split("_")[-1])
+    for i in range(0, np.size(lines_orig)):
+        if lines_orig[i][0] == "s":
+            year = int(lines_orig[i].split("\t")[2])
+
+            if year <= data_end_year:
+                line_ind.append(i)
+
+    # Remove excessive time points cross year bounds from 6 hourly data
+    end_ind = line_ind[-1]
+    lines = lines_orig[0:end_ind]
 
     # Calculate number of storms and max length
     num_storms, max_len = _calc_num_storms_and_max_len(lines)
     # Parse variables from TE stitch file
     te_stitch_vars = _get_vars_from_te_stitch(lines, max_len, num_storms)
+    # Add year info
+    te_stitch_vars["year_start"] = data_start_year
+    te_stitch_vars["year_end"] = data_end_year
+    te_stitch_vars["num_years"] = data_end_year - data_start_year + 1
```

Comment:
If I understand correctly, you moved the year info out of `_get_vars_from_te_stitch()` because they are now different after removing the excessive time points that cross year bounds. Is this right?

Reply:
The start and end years were previously based on data from te_stitches files. It worked fine earlier because in the simulated data, there are always TCs each year. It became problematic when testing v3 datasets, years with no TCs get skipped. The new code now use start and end included in file name, which represent the actually year range being assessed.